sys/net/gnrc/netif/gnrc_netif_device_type.c

```diff
@@ -0,0 +1,157 @@
+#if defined(MODULE_CC110X) || defined(MODULE_NRFMIN)
+static void _create_iid_from_short(const uint8_t *addr, size_t addr_len,
+                                   eui64_t *iid)
+{
+    const unsigned offset = sizeof(eui64_t) - addr_len;
+
+    memset(iid->uint8, 0, sizeof(iid->uint8));
+    iid->uint8[3] = 0xff;
+    iid->uint8[4] = 0xfe;
+    memcpy(&iid->uint8[offset], addr, addr_len);
+}
+#endif /* defined(MODULE_NETDEV_IEEE802154) || defined(MODULE_CC110X) ||
+        * defined(MODULE_NRFMIN) || defined(MODULE_XBEE) */
+
+int gnrc_netif_ipv6_iid_from_addr(const gnrc_netif_t *netif,
+                                  const uint8_t *addr, size_t addr_len,
+                                  eui64_t *iid)
+{
+#if GNRC_NETIF_L2ADDR_MAXLEN > 0
+    if (netif->flags & GNRC_NETIF_FLAGS_HAS_L2ADDR) {
+        switch (netif->device_type) {
+#if defined(MODULE_NETDEV_ETH) || defined(MODULE_ESP_NOW)
+            case NETDEV_TYPE_ETHERNET:
+            case NETDEV_TYPE_ESP_NOW:
+                if (addr_len == ETHERNET_ADDR_LEN) {
+                    ethernet_get_iid(iid, (uint8_t *)addr);
+                    return sizeof(eui64_t);
+                }
+                else {
+                    return -EINVAL;
+                }
+#endif  /* defined(MODULE_NETDEV_ETH) || defined(MODULE_ESP_NOW) */
+#if defined(MODULE_NETDEV_IEEE802154) || defined(MODULE_XBEE)
+            case NETDEV_TYPE_IEEE802154:
+                if (ieee802154_get_iid(iid, addr, addr_len) != NULL) {
+                    return sizeof(eui64_t);
+                }
+                else {
+                    return -EINVAL;
+                }
+#endif  /* defined(MODULE_NETDEV_IEEE802154) || defined(MODULE_XBEE) */
+#ifdef MODULE_NORDIC_SOFTDEVICE_BLE
+            case NETDEV_TYPE_BLE:
+                if (addr_len == sizeof(eui64_t)) {
+                    memcpy(iid, addr, sizeof(eui64_t));
+                    iid->uint8[0] ^= 0x02;
+                    return sizeof(eui64_t);
+                }
+                else {
+                    return -EINVAL;
+                }
+#endif  /* MODULE_NORDIC_SOFTDEVICE_BLE */
+#if defined(MODULE_CC110X) || defined(MODULE_NRFMIN)
+            case NETDEV_TYPE_CC110X:
+            case NETDEV_TYPE_NRFMIN:
+                if (addr_len <= 3) {
+                    _create_iid_from_short(addr, addr_len, iid);
+                    return sizeof(eui64_t);
+                }
+                else {
+                    return -EINVAL;
+                }
+#endif  /* defined(MODULE_CC110X) || defined(MODULE_NRFMIN) */
+            default:
+                (void)addr;
+                (void)addr_len;
+                (void)iid;
+#ifdef DEVELHELP
+                LOG_ERROR("gnrc_netif: can't convert hardware address to IID "
+                          "on interface %u\n", netif->pid);
+#endif  /* DEVELHELP */
+                assert(false);
+                break;
+        }
+    }
+#endif /* GNRC_NETIF_L2ADDR_MAXLEN > 0 */
+    return -ENOTSUP;
```

Comment:
this is never reached (every switch statement terminates, see the `assert(false)`). Is this wanted? So probably the doc for this functions needs to be updated...

Reply:
What @maribu is saying. Also it is in accordance with what you proposed ;-).